Wikipedia/Code/WMFRandomArticleFetcher.m

```diff
@@ -21,37 +21,64 @@ @implementation WMFRandomArticleFetcher
 - (instancetype)init {
     self = [super init];
     if (self) {
-        AFHTTPSessionManager *manager = [AFHTTPSessionManager wmf_createDefaultManager];
-        manager.responseSerializer = [WMFMantleJSONResponseSerializer serializerForValuesInDictionaryOfType:[MWKSearchResult class] fromKeypath:@"query.pages" emptyValueForJSONKeypathAllowed:NO];
-        self.operationManager = manager;
+        self.session = [WMFSession shared];
     }
     return self;
 }
 
-- (void)dealloc {
-    [self.operationManager invalidateSessionCancelingTasks:YES];
-}
-
-- (BOOL)isFetching {
-    return [[self.operationManager operationQueue] operationCount] > 0;
-}
+- (void)fetchRandomArticleWithSiteURL:(NSURL *)siteURL completion:(void (^)(NSError *_Nullable error, MWKSearchResult *_Nullable result))completion {
+    NSParameterAssert(siteURL);
+    if (siteURL == nil) {
+        NSError *error = [NSError wmf_errorWithType:WMFErrorTypeInvalidRequestParameters
+                                           userInfo:nil];
+        completion(error, nil);
+        return;
+    }
 
-- (void)fetchRandomArticleWithSiteURL:(NSURL *)siteURL failure:(nonnull WMFErrorHandler)failure success:(nonnull WMFMWKSearchResultHandler)success {
+    NSURL *url = [WMFConfiguration.current mediawikiAPIURLForHost:siteURL.host];
     NSDictionary *params = [[self class] params];
 
-    [self.operationManager wmf_apiZeroSafeGETWithURL:siteURL
-        parameters:params
-        success:^(NSURLSessionDataTask *operation, NSArray *responseObject) {
-            [[MWNetworkActivityIndicatorManager sharedManager] pop];
```

Comment:
does this need this `pop`? there was no matching `[[MWNetworkActivityIndicatorManager sharedManager] push]` before